### ISSUE TITLE
[Bug]: Config is read too early to be modified in test code

### ISSUE THREAD

**Issue by yiftachkarkason** [testcontainers/testcontainers-go#2636]
### Testcontainers version

0.32.0

### Using the latest Testcontainers version?

Yes

### Host OS

MacOS

### Host arch

ARM

### Go version

1.22.5

### Docker version

```shell
Client: Docker Engine - Community
 Version:           27.0.3
 API version:       1.45 (downgraded from 1.46)
 Go version:        go1.22.4
 Git commit:        7d4bcd863a
 Built:             Fri Jun 28 14:56:30 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.31.0 (153195)
 Engine:
  Version:          26.1.4
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.11
  Git commit:       de5c9cf
  Built:            Wed Jun  5 11:29:12 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.33
  GitCommit:        d2d58213f83a351ca8f528a95fbd145f5654e957
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Docker info

```shell
Client: Docker Engine - Community
 Version:    27.0.3
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.14.1-desktop.1
    Path:     /Users/username/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.27.1-desktop.1
    Path:     /Users/username/.docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.32
    Path:     /Users/username/.docker/cli-plugins/docker-debug
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     /Users/username/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.24
    Path:     /Users/username/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     /Users/username/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.2.0
    Path:     /Users/username/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/username/.docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.9.3
    Path:     /Users/username/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 112
 Server Version: 26.1.4
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d2d58213f83a351ca8f528a95fbd145f5654e957
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.6.31-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 14
 Total Memory: 7.754GiB
 Name: docker-desktop
 ID: 9d944c12-17c0-4536-a1e9-d2a39d3a288c
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=unix:///Users/username/Library/Containers/com.docker.docker/Data/docker-cli.sock
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### What happened?

[A recent change in version 0.32.0 introduced a `config.Read()` call in the package `init` function](https://github.com/testcontainers/testcontainers-go/commit/80e57e194b1bd9341198b485ece84e73116c4ee4#diff-ff87b7c4777a35588053a509583d66c9f404ccbea9e1c71d2a5f224d7ad1323eR31), which means that the environment variables and/or properties file is read very early when the program starts.

In my E2E tests, I set relevant environment variables (such as disabling Ryuk for the longer tests to avoid having the containers removed while the test is still running), and this change breaks this functionality.
Currently, I can't upgrade the library past v0.31.0.

AFAICT there are 2 possible solutions:
1. Avoid calling `config.Read` during the package `init` function
2. Expose the `config.Reset` function to external callers by extracting it from the `internal` directory.

Thank you for you help!

**Comment by stevenh** [comment]
I tripped up against this too, ready about using the env tried it and saw exactly what you mentioned making env variables unusable in tests.

**Comment by mdelapenya** [comment]
Indeed. I want to definitively fix the config tests that depends a lot in the current state to make the tests deterministic. And, to the matter of this issue, look for a better mechanism to initialise the verbosity of the ryuk message. I _think_ we could move that log entry to the place Ryuk is called for first time, so that the init function of that package does not interfere with the config.Read. Thoughts?

**Comment by yiftachkarkason** [comment]
Yeah, that would be better for my use case because I'm configuring Ryuk the same for all tests.
I would still consider exposing a "ReloadConfig" function to allow users to overcome such hurdles independently.

**Comment by mdelapenya** [comment]
@yiftachkarkason we released `v0.33.0` Could you please check with that version?

If fixed, I think we can close this one, thanks!

**Comment by yiftachkarkason** [comment]
Tested in v0.33.0 and it looks good! Thank you 😄
